Setting Screen/Sources/View/SettingViewController.swift

```diff
@@ -0,0 +1,150 @@
+        search.
+        return search
+    }()
+
+    private lazy var tableView: UITableView = {
+        let tableView = UITableView(frame: .zero, style: .insetGrouped)
+        tableView.register(StandartTableViewCell.self, forCellReuseIdentifier: "standartCell")
+        tableView.register(UserInfoTableViewCell.self, forCellReuseIdentifier: "userInfoCell")
+        tableView.register(FamilyInfoTableViewCell.self, forCellReuseIdentifier: "familyInfoCell")
+        tableView.dataSource = self
+        tableView.delegate = self
+        return tableView
+    }()
+
+    // MARK: - Lifecycle
+
+    override func viewWillAppear(_ animated: Bool) {
+        tableView.reloadData()
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        title = "Настройки"
+        navigationController?.navigationBar.prefersLargeTitles = true
+        view.backgroundColor = .systemGray6
+        setupHierarchy()
+        setupLayout()
+
+        settingCells = Setting.settingCells
+    }
+
+    // MARK: - Setup
+
+    private func setupHierarchy() {
+        navigationItem.searchController = search
+        view.addSubview(tableView)
+    }
+
+    private func setupLayout() {
+        tableView.snp.makeConstraints { make in
+            make.top.left.right.bottom.equalTo(view)
+        }
+    }
+
+    // MARK: - Actions
+
+}
+
+extension SettingViewController: UITableViewDataSource, UITableViewDelegate {
+    func numberOfSections(in tableView: UITableView) -> Int {
+        return settingCells?.count ?? 0
+    }
+
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return settingCells?[section].count ?? 0
+    }
+
+    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        guard let cellType = settingCells?[indexPath.section][indexPath.row].type else { return 44 }
+
+        switch cellType {
+        case .userInfo:
+            return 90
+        case .familyInfo:
+            return 48
+        case .standart, .cellWithSwitch:
+            return 44
+        }
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        guard let cell = settingCells?[indexPath.section][indexPath.row] else { return UITableViewCell() }
+        return configureCell(cell: cell, indexPath: indexPath)
+    }
+
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        tableView.deselectRow(at: indexPath, animated: true)
+        guard let cell = settingCells?[indexPath.section][indexPath.row] else { return }
+        print("Вы нажали \(cell.title)")
+
+        switch cell.type {
+        case .cellWithSwitch:
+            return
+        case .userInfo, .familyInfo, .standart:
+            let detailViewController = DetailViewController()
+            detailViewController.settingCell = settingCells?[indexPath.section][indexPath.row]
```

Comment:
Settings у DetailView можно сделать приватной и передавать через инициализатор, а не обращаться к этому свойству из вне. Тем более у нас в любом случае создается новый экземпляр DetailView. Будет выглядеть примерно так
```suggestion
            let settings = settingCells?[indexPath.section][indexPath.row]
            let detailViewController = DetailViewController(settings: settings)
```